.github/workflows/ci.yml

```diff
@@ -0,0 +1,29 @@
+name: CI
+on:
+  workflow_dispatch:
+  pull_request:
+
+  # triggering CI default branch improves caching
+  # see https://docs.github.com/en/free-pro-team@latest/actions/guides/caching-dependencies-to-speed-up-workflows#restrictions-for-accessing-a-cache
+  push:
+    branches:
+      - main
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup JDK
+        uses: actions/setup-java@v4
+        with:
+          distribution: corretto
+          java-version: 11
+          cache: sbt
+      - name: Build and Test
+        run: sbt -v +test
+      - uses: EnricoMi/publish-unit-test-result-action@v1
```

Comment:
I haven't personally used https://github.com/EnricoMi/publish-unit-test-result-action before, but I see that it is used in other Content Pipeline repos, like https://github.com/guardian/content-api-scala-client/blob/ed01efc55163e072601012a205861ce5467d5a5e/.github/workflows/testonpush.yml#L34

Previously, I've used https://github.com/test-summary/action, which I think is maybe more limited? Happy to try something new!